content/general/concepts/big-o-notation/big-o-notation.md

```diff
@@ -0,0 +1,45 @@
+---
+Title: 'Big-O Notation'
+Description: 'Big-O notation is a form of measuring the complexity of an algorithm. '
+Subjects:
+  - 'Computer Science'
+  - 'Web Development'
+Tags:
+  - 'Algorithms'
+CatalogContent:
+  - 'paths/computer-science'
+  - 'paths/front-end-engineer-career-path'
+---
+
+**Big-O notation** is a form of measuring the complexity of an algorithm. This can take the form of a time complexity analysis or a space complexity analysis. At times, an algorithm can be refactored or reworked to achieve a more efficient time and space complexity. To describe the time or space complexity, two different notations are used: big omega notation and big-O. Big omega describes the best running case scenario while big-O describes the worst case scenario.
+
+## Common Runtimes
+
+Here are some common runtimes:
+
+![graft of the different kinds of big O notation](https://raw.githubusercontent.com/Codecademy/docs/main/media/bigo-graph.png)
+
+- Polynomial time (red) $O(n^{2})$: If the time is a power of the input size.
+  - Bubble sort has quadratic time complexity.
+- Linear time (blue) $O(n)$: If the time is proportional to the input size.
+  - The traverse of a list.
+- Logarithmic time (green) $O(ln(n))$: If the time is a logarithmic function of the input size.
+  - Binary search algorithm.
+- Constant time (purple) $O(1)$: If the time needed by the algorithm is the same, regardless of the input size.
+  - An access to an array element.
```

Comment:
```suggestion
| Runtime | Name | Description | Use Case(s) |
| :---: | :---: | --- | --- |
| $O(n^{2})$ | Polynomial/quadratic time (in red) | The time needed is the input size, `n`, multiplied by itself. | [Bubble sort](https://www.codecademy.com/learn/sorting-algorithms-java/modules/bubble-sort-java/cheatsheet) has quadratic time complexity. |
| $O(n)$ | Linear time (in blue) | The time needed is proportional to the input size `n`. | Traversing an [array](https://www.codecademy.com/resources/docs/general/array) of `n`-size. |
| $O(ln(n))$ | Logarithmic time (in green) | The time is a logarithmic function `ln()` of the input size `n`. | Analyzing the [binary search algorithm](https://www.codecademy.com/learn/fscp-22-search-graph-search-algorithms/modules/wdcp-22-binary-search-and-search-trees/cheatsheet). |
| $O(1)$ | Constant time (in purple) | If the time needed by the algorithm is the same, regardless of the input size. | Accessing an element in an array. |
```

I think this kind of information looks better in a markdown table as opposed to a list. Also, I admire your use of [LaTeX shorthands in GH Markdown](https://github.blog/2022-05-19-math-support-in-markdown/#:~:text=Support%20for%20displaying%20math%20expressions,the%20highly%20popular%20MathJax%20library.). I never heard of them until I reviewed your contribution! 😄

One question I had for the example used under "Logarithmic time" was that the [source](https://www.codecademy.com/learn/fscp-22-search-graph-search-algorithms/modules/wdcp-22-binary-search-and-search-trees/cheatsheet) that I linked reports the runtime for binary search as $O(log(n))$, not $O(ln(n))$. Just wanted to confirm whether something needed to be changed based on this info?

cc: @SSwiniarski